src/coreclrDebug/activate.ts

```diff
@@ -278,6 +281,14 @@ export class DebugAdapterExecutableFactory implements vscode.DebugAdapterDescrip
         // use the executable specified in the package.json if it exists or determine it based on some other information (e.g. the session)
         if (!executable) {
             const dotNetInfo = await getDotnetInfo(omnisharpOptions.dotNetCliPaths);
+            const hostExecutableResolver = new DotnetRuntimeExtensionResolver(
```

Comment:
Note - this currently *only* cases about the installed runtimes.  It does not confirm there is an SDK.  It will also only install a 7.0 runtime.  Is that what you expect?

Reply:
I _think_ this is the best we have for now until the .NET SDK installation API is avaliable to use.

This is to just workaround the scenario where dotnet exists in the PATH and we can hope there is an SDK.

Reply:
Do you require a .net 7 SDK or higher?  Or can you use a .net 6 SDK?  Because if oyu only have a .net 6 SDK installed on the path, this will return a locally installed .net 7 runtime (with no SDK information, you won't be able to see the .net 6 SDK from it)

Reply:
What Andrew is trying to do is find a version of .NET for the target app. The debugger itself will ignore `DOTNET_ROOT` and run on its own runtime.

Reply:
It would be whatever the user's target framework would be. 

The scenario is that the user was able to build their application with `dotnet build` and I was hoping this API would help me grab the dotnet install path to set as DOTNET_ROOT.

E.g. `~/.dotnet`

Reply:
So if the user's app targets .net6 and they only have .net6 SDK installed, this API *will not* return the path to the dotnet 6 installation.  It will return a locally installed .net7 runtime path.  Just want to make sure you know what to expect from this API.

Reply:
See https://github.com/dotnet/vscode-csharp/pull/6567/files#r1364688699

Maybe instead what should happen is `getDotnetInfo` should return the *real* CLI path *and* the set of SDKs that are installed at that path (via `dotnet --list-sdks`) - and then you use that.

Because `DotnetRuntimeExtensionResolver` is really focused on making sure the runtime to host the server processes is available.